OpenSim/Common/ComponentPath.h

```diff
@@ -0,0 +1,90 @@
+ * Copyright (c) 2005-2016 Stanford University and the Authors                *
+ * Author(s): Carmichael Ong                                                  *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+#include "Path.h"
+
+namespace OpenSim {
+
+//==============================================================================
+//                            OPENSIM COMPONENT PATH
+//==============================================================================
+/**
+* A class for handling Paths for Components, deriving from Path. A ComponentPath
+* uses a forward-slash ('/') as a separator always. It also specifies invalid
+* characters in a Component name:
+* - back-slash ('\\')
+* - forward-slash ('/')
+* - asterisk ('*')
+* - plus-sign ('+')
+*
+* @author Carmichael Ong
+*/
+
+class OSIMCOMMON_API ComponentPath : public Path {
+    OpenSim_DECLARE_CONCRETE_OBJECT(ComponentPath, Path);
+
+public:
+    /// Constructors
+    ComponentPath() = default;
+    ComponentPath(const std::string path);
+    ComponentPath(std::vector<std::string> pathVec, bool isAbsolute);
+
+    /// Operators
+    bool operator==(const ComponentPath& other) const
+    {
+        return this->getString() == other.getString();
+    }
+
+    bool operator!=(const ComponentPath& other) const
+    {
+        return this->getString() != other.getString();
+    }
+
+    /// Get an absolute path by resolving it relative to a given otherPath.
+    /// If the current Path is already absolute, return the same Path.
+    ComponentPath getAbsolutePath(ComponentPath* otherPath);
+
+    /// Find the relative Path between this Path and another Path (otherPath)
+    /// (i.e. the Path to go FROM otherPath TO this Path). Both Paths must be 
+    /// absolute.
+    ComponentPath getRelativePath(ComponentPath* otherPath);
+
+    /// Return the the sub-path that contains all sub components except for
+    /// the last one.
+    ComponentPath getParentPath();
+
+    /// Return a ComponentPath object of a subcomponent by index. This will
+    /// always be a relative path.
+    ComponentPath getSubcomponent(size_t index);
```

Comment:
Sounds like the function name and comments are unclear. As an example:
`ComponentPath path("/a/b/c/d");`
`ComponentPath subcomponentPath(2);`
`std::string str = subcomponentPath.getString(); // this returns "c" specifically not "/c"`
`subcomponentPath.isAbsolute(); // this returns false`

The purpose of this method was to walk through and compare parts of a path, as seen [here](https://github.com/opensim-org/opensim-core/pull/1215/files#diff-a8c6cc5aa99c4c61fcc4d4b0a45a4813R2095).